x-pack/plugins/event_log/server/es/cluster_client_adapter.ts

```diff
@@ -238,41 +248,123 @@ export class ClusterClientAdapter<TDoc extends { body: AliasAny; index: string }
+      },
+      {
+        term: {
+          'kibana.saved_objects.type': {
+            value: type,
+          },
+        },
+      },
+      // @ts-expect-error undefined is not assignable as QueryDslTermQuery value
+      namespaceQuery,
+    ];
+
     const musts: estypes.QueryDslQueryContainer[] = [
       {
         nested: {
           path: 'kibana.saved_objects',
           query: {
             bool: {
-              must: [
-                {
-                  term: {
-                    'kibana.saved_objects.rel': {
-                      value: SAVED_OBJECT_REL_PRIMARY,
+              must: reject(savedObjectsQueryMust, isUndefined),
+            },
+          },
+        },
+      },
+    ];
+
+    const shouldQuery = [];
+
+    shouldQuery.push({
+      bool: {
+        must: [
+          {
+            nested: {
+              path: 'kibana.saved_objects',
+              query: {
+                bool: {
+                  must: [
+                    {
+                      terms: {
+                        // default maximum of 65,536 terms, configurable by index.max_terms_count
+                        'kibana.saved_objects.id': ids,
+                      },
                     },
+                  ],
+                },
+              },
+            },
+          },
+          {
+            range: {
+              'kibana.version': {
+                gte: LEGACY_ID_CUTOFF_VERSION,
+              },
+            },
+          },
+        ],
+      },
+    });
+
+    if (legacyIds && legacyIds.length > 0) {
+      shouldQuery.push({
+        bool: {
+          must: [
+            {
+              nested: {
+                path: 'kibana.saved_objects',
+                query: {
+                  bool: {
+                    must: [
+                      {
+                        terms: {
+                          // default maximum of 65,536 terms, configurable by index.max_terms_count
+                          'kibana.saved_objects.id': legacyIds,
+                        },
+                      },
+                    ],
                   },
                 },
-                {
-                  term: {
-                    'kibana.saved_objects.type': {
-                      value: type,
+              },
+            },
+            {
+              bool: {
+                should: [
+                  {
+                    range: {
+                      'kibana.version': {
+                        lt: LEGACY_ID_CUTOFF_VERSION,
+                      },
                     },
                   },
-                },
-                {
-                  terms: {
-                    // default maximum of 65,536 terms, configurable by index.max_terms_count
-                    'kibana.saved_objects.id': ids,
+                  {
+                    term: {
+                      'kibana.version': {
+                        value: 'NULL',
+                      },
+                    },
```

Comment:
I had to change it to what's below to get the failing test to pass. Will require updates to the jest tests after as well.

```suggestion
                    bool: {
                      must_not: {
                        exists: {
                          field: 'kibana.version',
                        },
                      },
                    },
```